internal/pkg/cli/storage_init.go

```diff
@@ -724,6 +724,13 @@ func (o *initStorageOpts) validateOrAskAuroraInitialDBName() error {
 
 // Execute deploys a new environment with CloudFormation and adds it to SSM.
 func (o *initStorageOpts) Execute() error {
+	if o.addIngressFrom != "" {
```

Comment:
I wonder if it's worthwhile encapsulating this in a separate method, something like:
```go
func (o *initStorageOpts) Execute() error {
    o.consumeFlags()
}
```

Reply:
As a larger question, I assume our flow should be like this:
```
Validate() // Never sets any internal state and just checks if flags are compatible.
Ask() // Sets only flag values in the `*Vars` struct
Execute() // Sets all internal state
```

Reply:
There is one internal state that gets set during `Validate` and `Ask`: `o.workloadExists`. The intention was to reduce function calls.

However, the cost of that function call isn't large though (one more fs read), so the pros brought by caching internal state in `Validate`/`Ask` (thus going against the pattern) is marginal. So I changed the implementation to ⬇️ 

https://github.com/aws/copilot-cli/pull/4479/commits/3a0bde20c82f5d5ae7943deb65087ae7b51799c7